hivemq-edge/src/main/java/com/hivemq/api/auth/provider/impl/ldap/LdapClient.java

```diff
@@ -0,0 +1,202 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.api.auth.provider.impl.ldap;
+
+import com.unboundid.ldap.sdk.BindRequest;
+import com.unboundid.ldap.sdk.BindResult;
+import com.unboundid.ldap.sdk.LDAPConnection;
+import com.unboundid.ldap.sdk.LDAPConnectionPool;
+import com.unboundid.ldap.sdk.LDAPException;
+import com.unboundid.ldap.sdk.ResultCode;
+import com.unboundid.ldap.sdk.SimpleBindRequest;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.GeneralSecurityException;
+
+/**
+ * LDAP client that manages connections and provides authentication operations.
+ * <p>
+ * This class wraps the UnboundID LDAP SDK and provides a simplified API for LDAP operations
+ * with proper lifecycle management and connection pooling.
+ * <p>
+ * Usage:
+ * <pre>{@code
+ * LdapClient client = new LdapClient(connectionProperties);
+ * client.start();
+ * try {
+ *     boolean authenticated = client.bindUser(userDn, password);
+ *     // ... use client
+ * } finally {
+ *     client.stop();
+ * }
+ * }</pre>
+ */
+public class LdapClient {
+
+    private static final @NotNull Logger log = LoggerFactory.getLogger(LdapClient.class);
+
+    private final @NotNull LdapConnectionProperties connectionProperties;
+    private final @NotNull UserDnResolver userDnResolver;
+    private volatile LDAPConnectionPool connectionPool;
+    private volatile boolean started = false;
+
+    /**
+     * Creates a new LDAP client with the specified connection properties.
+     *
+     * @param connectionProperties The connection configuration
+     */
+    public LdapClient(final @NotNull LdapConnectionProperties connectionProperties) {
+        this.connectionProperties = connectionProperties;
+        this.userDnResolver = connectionProperties.createUserDnResolver();
+    }
+
+    /**
+     * Starts the LDAP client and initializes the connection pool.
+     *
+     * @throws LDAPException            if the connection pool cannot be created
+     * @throws GeneralSecurityException if there's an SSL/TLS configuration issue
+     * @throws IllegalStateException    if the client is already started
+     */
+    public synchronized void start() throws LDAPException, GeneralSecurityException {
+        if (started) {
+            throw new IllegalStateException("LDAP client is already started");
+        }
+
+        log.debug("Starting LDAP client, connecting to {}:{}",
```

Comment:
there are a number of `log.debug` statements without the isDebugEnabled guard that I have observed in this class.